crates/router/src/core/payment_methods/cards.rs

```
@@ -403,34 +406,107 @@ pub async fn migrate_payment_method(
-            .await
+            .await?
         }
-    }
+    };
+    let payment_method_response = match resp {
+        services::ApplicationResponse::Json(response) => response,
+        _ => Err(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to fetch the payment method response")?,
+    };
+
+    let pm_id = payment_method_response.payment_method_id.clone();
+
+    let network_token = req.network_token.clone();
+
+    let network_token_validation = network_token.clone().map(|network_token| {
+        cards::CardNumber::from_str(network_token.network_token_data.network_token_number.peek())
+            .map_err(|_| false)
+    }); // network_token_migrated is true if network_token_number passes the validation and will be migrated to locker
+
+    let network_token_migrated = match network_token_validation {
+        Some(Ok(network_token)) => {
+            logger::debug!("Network token to be migrated is valid, saving in locker");
+            let network_token_details = req
+                .network_token
+                .clone()
+                .get_required_value("network_token")?;
+            let network_token_requestor_ref_id =
+                network_token_details.network_token_requestor_ref_id.clone();
+            let network_token_data = network_token_details.network_token_data.clone();
+
+            Some(
+                save_network_token_and_update_payment_method(
+                    state,
+                    &req,
+                    key_store,
+                    merchant_account,
+                    network_token_data,
+                    network_token_requestor_ref_id,
+                    network_token,
+                    pm_id,
+                )
+                .await
+                .map_err(|err| logger::error!(?err, "Failed to save network token in locker"))
+                .ok()
+                .unwrap_or_default(),
+            )
+        }
+        Some(Err(_)) => {
```

Don't discard the error, log the error